zerver/tests/test_message_delete.py

```diff
@@ -573,6 +578,53 @@ def check_delete_message_by_other_user(
         check_delete_message_by_other_user(
             "cordelia", "polonius", "You don't have permission to delete this message"
         )
+        check_delete_message_by_other_user(
+            "cordelia", "shiva", "You don't have permission to delete this message"
+        )
+
+        do_change_stream_group_based_setting(
```

Comment:
I think it would be good to test stream-level setting in isolation by setting realm level setting to "Nobody" group.